cognite/extractorutils/uploader/upload_failure_handler.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+from typing import Any, Iterator, List
+
+import jsonlines
+
+
+class FileErrorMapping:
+    def __init__(self, file_name: str, error_reason: str) -> None:
+        self.file_name = file_name
+        self.error_reason = error_reason
+
+    def __iter__(self) -> Iterator[List[str]]:
+        return iter([[self.file_name, self.error_reason]])
+
+
+class FileFailureManager:
+    MAX_QUEUE_SIZE = 500
+    START_TIME_KEY = "start_time"
+    FILE_REASON_MAP_KEY = "file_error_reason_map"
+
+    def __init__(self, start_time: str | None = None, path_to_file: str | None = None) -> None:
+        self.failure_logs: dict[str, Any] = {}
```

Comment:
I don't really understand this implementation. From what I can tell, `failure_logs` only contains two fields `start_time` or `file_error_reason_map`. Why are these not attributes on the class instead of being hidden inside a dictionary without any typing?

Reply:
@einarmo So firstly I had to make it `Any` mainly because it causes issues while calling methods like `.update` which I call at a later point.

Also do you mean I should maintain it separately like `self.start_time` and `self.logs`. And before writing to the file just dynamically create a dictionary combining the logs and timestamp.

Reply:
Yeah, that would make better code, I think.

Reply:
okay that is fair. Fixed that change up @einarmo. let me know how it looks. 
Thanks in advance